python/tvm/driver/tvmc/common.py

```diff
@@ -17,6 +17,74 @@
 """
 Common utility functions shared by TVMC modules.
 """
+import argparse
+import re
+
+from tvm import relay
+from tvm import transform
 
 class TVMCException(Exception):
     """TVMC Exception"""
+
+
+def convert_graph_layout(mod, desired_layout):
```

Comment:
ConvertLayout pass doesn't check if the input and desired layouts are the same, so you need to make sure it won't be a problem. Please cover it in the unit test.

Reply:
Good catch, I didn't know that it didn't check for layouts, but apparently it won't add anything if it is not needed. So I wrote a few tests and even raised #6410 for and issue that was happening.

The new version includes these fixes and tests.